enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTest.java

```diff
@@ -165,925 +111,579 @@ public void shouldChangeUserPasswordIfSameUser() throws Exception
-    Henrik starts write transaction → ok
-    Henrik starts read transaction → ok
-    Henrik starts schema transaction → ok
-    Henrik creates user Craig → permission denied
-    Henrik logs off
-    */
+    TODO: uncomment and fix
     @Test
-    public void userCreation4() throws Exception
+    public void shouldNotAllowDeletingYourself() throws Exception
     {
-        testCallEmpty( db, adminSubject, "CALL dbms.createUser('Henrik', 'bar', false)" );
-        AuthSubject subject = manager.login( authToken( "Henrik", "bar" ) );
-        assertEquals( AuthenticationResult.SUCCESS, subject.getAuthenticationResult() );
-        testFailReadAction( subject, 3 );
-        testFailWriteAction( subject );
-        testFailSchema( subject );
-        testFailCreateUser( subject );
-        testCallEmpty( db, adminSubject, "CALL dbms.addUserToRole('Henrik', '" + ARCHITECT + "')" );
-        testSuccessfulWriteAction( subject );
-        testSuccessfulReadAction( subject, 4 );
-        testSuccessfulSchemaAction( subject );
-        testFailCreateUser( subject );
+        testCallFail( adminSubject, "CALL dbms.deleteUser('" + adminSubject.name() + "')",
+                QueryExecutionException.class, "Deleting yourself is not allowed" );
     }
+    */
 
-    //----------User and role management-----------
-    @Test
-    public void shouldAllowAddingAndRemovingUserFromRole() throws Exception
-    {
-        assertFalse( "Should not have role publisher",
-                ShiroAuthSubject.castOrFail( readSubject ).getSubject().hasRole( PUBLISHER ) );
-        testCallEmpty( db, adminSubject, "CALL dbms.addUserToRole('readSubject', '" + PUBLISHER + "')" );
-        assertTrue( "Should have role publisher",
-                ShiroAuthSubject.castOrFail( readSubject ).getSubject().hasRole( PUBLISHER ) );
-        testCallEmpty( db, adminSubject, "CALL dbms.removeUserFromRole('readSubject', '" + PUBLISHER + "')" );
-        assertFalse( "Should not have role publisher",
-                ShiroAuthSubject.castOrFail( readSubject ).getSubject().hasRole( PUBLISHER ) );
-    }
+    //---------- suspend user -----------
 
     @Test
-    public void shouldAllowAddingUserToRoleMultipleTimes() throws Exception
+    public void shouldSuspendUser() throws Exception
     {
-        assertFalse( "Should not have role publisher",
-                ShiroAuthSubject.castOrFail( readSubject ).getSubject().hasRole( PUBLISHER ) );
-        testCallEmpty( db, adminSubject, "CALL dbms.addUserToRole('readSubject', '" + PUBLISHER + "')" );
-        testCallEmpty( db, adminSubject, "CALL dbms.addUserToRole('readSubject', '" + PUBLISHER + "')" );
-        assertTrue( "Should have role publisher",
-                ShiroAuthSubject.castOrFail( readSubject ).getSubject().hasRole( PUBLISHER ) );
+        testCallEmpty( adminSubject, "CALL dbms.suspendUser('readSubject')" );
+        assertTrue( manager.getUser( "readSubject" ).hasFlag( FileUserRealm.IS_SUSPENDED ) );
```

Comment:
So we actually do this in the first suspend scenario. My plan was to keep stuff minimal in the AuthProceduresTest, and have more complex interactions (using multiple procedures or actions) in the AuthScenariosIT